domain/consensus/processes/coinbasemanager/coinbasemanager.go

```diff
@@ -95,6 +116,31 @@ func (c *coinbaseManager) coinbaseOutputForBlueBlock(blueBlock *externalapi.Doma
 	return txOut, true, nil
 }
 
+func (c *coinbaseManager) coinbaseOutputForRewardFromRedBlocks(ghostdagData *model.BlockGHOSTDAGData,
+	acceptanceData externalapi.AcceptanceData, daaAddedBlocksSet hashset.HashSet,
+	coinbaseData *externalapi.DomainCoinbaseData) (*externalapi.DomainTransactionOutput, bool, error) {
+
+	totalReward := uint64(0)
+	mergeSetBluesCount := len(ghostdagData.MergeSetBlues())
+	for i, red := range ghostdagData.MergeSetReds() {
+		reward, err := c.calcMergedBlockReward(red, acceptanceData[mergeSetBluesCount+i], daaAddedBlocksSet)
```

Comment:
We need to be careful with this, if we ever change the ordering of the acceptance data(ie https://github.com/kaspanet/kaspad/issues/1558)